sm/json_sm/src/error.rs

```diff
@@ -64,6 +64,13 @@ pub enum SoftwareError {
         name: SoftwareName,
         version: SoftwareVersion,
     },
+
+    #[error("The configured default plugin: {0} not found")]
+    InvalidDefaultPlugin(String),
+
+    #[error("I/O error: {reason:?}")]
+    IoError { reason: String },
+
```

Comment:
(minor) What has been done in the other modules is to make the IoError transparent:

```suggestion
    #[error(transparent)]
    IoError(#[from] std::io::Error),
```

Reply:
Yeah I had tried that, but it wasn't possible because this `SoftwareError` enum is marked as `Serialize`, `Deserialize`, `Clone` etc and  since `std::io::Error` isn't any of these, this wrapping was the only way I could find. Is there some other trick to conform to those requirements with `thiserror`?